utils.c

```diff
@@ -100,7 +100,7 @@ char * read_file(const char *file_name, size_t *p_size) {
 	*p_size = ftell(f);
 	char * buf = malloc(*p_size);
 	fseek(f, 0, SEEK_SET);
-	fread(buf, *p_size, 1, f);
+	if (fread(buf, *p_size, 1, f) != 1) { fprintf(stderr, "error durring fread"); };
```

Comment:
It would probably be better to change this so it becomes:
```
if (fread(buf, *p_size, 1, f) != 1) {
    fprintf(stderr, "error durring fread\n");
    exit(-1);
}
```